### DISCUSSION TITLE
Problems using the TypeScript SDK in Cloudflare Worker

### DISCUSSION THREAD

**Discussion by kulla** [nmshd/feedback#10] [General]
For fast prototyping we wanted to use [Cloudflare workers](https://developers.cloudflare.com/workers/) which should speak to the enmeshed connector. However since the [TypeScript SDK](https://github.com/nmshd/cns-connector/tree/main/packages/sdk) uses [axios](https://www.npmjs.com/package/axios) it cannot be used inside a Cloudflare worker since it only supports the fetch API (see [axios#2507](https://github.com/axios/axios/issues/2507#issuecomment-555971088) and [axios#1219](https://github.com/axios/axios/issues/1219)). There are problems reported with NextJS and Amazon AWS as well. However for the next version of axios (see [axios#4209](https://github.com/axios/axios/issues/4209)) a fetch-based adapter might be implemented (see [trello board](https://trello.com/b/SBqOxCst/axios-next) -> fetch based adapters are "Potential issues to be added"). I had a look into the SDK and I see some workarounds:

* Using [node-fetch](https://www.npmjs.com/package/node-fetch) or [isomorphic-fetch](https://www.npmjs.com/package/isomorphic-fetch) instead of axios.
* Have a abstract class `HttpClient` similar to `AxiosInstance` with only the methods which are actually used by the SDK. Now there could be a `FetchHttpClient` and `AxiosHttpClient`. Consumers can decide which of both they want to use.
* Wait until axios supports fetch (maybe it will be in the next version already).
* ...

In our case it is a low priority issue. For accessing the enmeshed connector we will use a dedicated micro service running in its own docker container so that we will not have this issue. That's also the reason why I haven't created a feature request. However I wanted to document this somewhere. So other developers will not get the same error as we did.

**Comment by jkoenig134** [comment]
Thanks, we will look into.

Waiting for axios to fix this seems to be a good option, we keep the dependencies up to date anyway!
